test/ring/middleware/test/anti_forgery.clj

```diff
@@ -68,6 +68,22 @@
       (is (= (get-in response [:session ::af/anti-forgery-token])
              (:body response))))))
 
+(deftest token-in-request-test
+ (letfn [(handler [request & more]
```

Comment:
This handler is incorrect. The 1 argument form is synchronous and the response is returned from the function. The 3 argument form is asynchronous and the response is passed to a callback.

It should read something more like:

```clojure
(letfn [(handler
          ([req] {:status 200, :headers {}, :body (:anti-forgery-token req)})
          ([req resp _] (resp (handler req))))]
```